### PR TITLE
Refactor AI_PROVIDER Handling in index.js

### PR DESCRIPTION
**Pull request by ohneda**
Hi,

I have been considering an implementation to specify a model for OpenAI's API. However, before proceeding with that, I realized it would be more efficient to split the handling of 'openai' and 'ollama' into separate files. 

## Changes made

Separation of AI_PROVIDER Logic: The conditional logic in index.js that handles different AI providers (openai and ollama) has been extracted into separate files.  This change ensures that adding a new provider in the future only requires creating a new file for it, without altering the existing codebase significantly.

This is not BREAKING CHANGE

Thanks,

### CODE REVIEW

**Comment by jaroslaw-weber** [comment]
nice refactor. im not a maintainer but probably good to merge after testing if it works well.

**Review by insulineru — APPROVED** [review]
Nice changes! Thanks for that!

**Pull request merged**
